File: client/producer.js
import { Kafka, Partitioners } from 'kafkajs';

const kafka = new Kafka({
    brokers: ['localhost:9092']
});

const producer = kafka.producer({
    createPartitioner: Partitioners.DefaultPartitioner
});

await producer.connect();

for (let i = 0; i < 10; i++) {
    const key = `key-${i}`;
    const value = `value-${i}`;
    await producer.send({
        topic: 'belajarkafka',
        messages: [
            { key, value }
        ]
    });
    console.log(`Sent message with key: ${key}, value: ${value}`);
}

await producer.disconnect();

File: client/python/producer.py
from kafka import KafkaProducer

producer = KafkaProducer(
    bootstrap_servers='localhost:9092'
)

for i in range(10):
    key = f'key-{i}'.encode()
    value = f'value-{i}'.encode()
    producer.send('belajarkafka', key=key, value=value)
    print(f"Sent message with key: {key.decode()}, value: {value.decode()}")

producer.flush()
producer.close()